pages/about/index.php

```diff
@@ -21,12 +21,10 @@
     case 'editorialTeam':
     case 'submissions':
     case 'contact':
-        define('HANDLER_CLASS', 'AboutContextHandler');
-        import('lib.pkp.pages.about.AboutContextHandler');
+        define('HANDLER_CLASS', '\\PKP\\pages\\about\\AboutContextHandler');
         break;
     case 'privacy':
     case 'aboutThisPublishingSystem':
-        define('HANDLER_CLASS', 'AboutSiteHandler');
-        import('lib.pkp.pages.about.AboutSiteHandler');
+        define('HANDLER_CLASS', '\\PKP\\pages\\about\\AboutSiteHandler');
```

Comment:
Just curious, is it possible for us to use the `::class` property here? Like:

```php
use PKP\pages\about\AboutSiteHandler;

...

define('HANDLER_CLASS', AboutSiteHandler::class);
```

Reply:
I'd like to introduce a different convention that does away with `HANDLER_CLASS` entirely; using defined constants for this stuff is weird to say the least (and dates back to PHP4 support). Let me think a little further about how far I'd like to push that for 3.4.0.

Reply:
`disable_path_info` is deprecated and I don't doubt it's already broken in some cases. I've started warning users e.g. on the forum that they should move away from this configuration and at some point we might need to write a script of some kind that redirects `disable_path_info`-style URLs to their nicer cousins.

You're right about router convergence, and I'm hoping at some point that there's a PSR that standardizes this before we try to replace our homebrew routing toolset.  So I'm thinking about this but hoping to wait on implementation for just a bit.

Reply:
:+1: Looks like the closest candidate may be [thephpleague/route](https://github.com/thephpleague/route)? It uses [FastRoute](https://github.com/nikic/FastRoute) under-the-hood, which Laravel's micro-framework Lumen also uses.

I think the big question is whether we will go all in with Laravel's Eloquent models. If so, their route-to-controller setup supports some automatic dependency injection (like knowing what object to get out the database based on a route) which might be useful. If not, a router that conforms to existing PSRs is best.

Reply:
I might experiment with `FastRoute` for some Swordv3 work -- it'll be good to know how well it interacts with everything else (Slim, our toolset, Laravel etc). I would love at some point to reconcile our roadmap WRT Slim and Laravel -- if we can land on something like FastRoute, it would make me very happy. (Our Slim library is currently out of date and it's one of the outstanding dependencies to review/resolve.)

Reply:
I've added a small change to the page handler code to avoid `HANDLER_CLASS`, without diverging too much from our current page handler practices. Essentially the page handler `index.php` wrappers can now instantiate and return handler classes directly, rather than having to define `HANDLER_CLASS`. See https://github.com/pkp/pkp-lib/issues/6091#issuecomment-796314189 for details.

I also considered having them pass back `SomeHandlerClass::class` rather than instantiating directly, but plugin handlers often don't share the same constructor signatures as page handlers in the applications, so this felt more flexible.

This is a small change but at least a step towards the patterns used by commodity routing frameworks.